tests/unit/utils/converter_unittest.cc

```diff
@@ -0,0 +1,78 @@
+#include <gtest/gtest.h>
+
+#include "sg_gtest_utilities.h"
+
+#include <shogun/lib/sg_types.h>
+#include <shogun/util/converters.h>
+
+#include <cstdint>
+#include <iostream>
+#include <limits>
+
+using namespace shogun;
+
+template <typename T>
+class Converters : public ::testing::Test
+{
+};
+
+SG_TYPED_TEST_CASE(Converters, sg_non_complex_types);
+
+TYPED_TEST(Converters, canDoCorrectConvertionOfUnsigned)
+{
+	TypeParam result = 0;
+	uint64_t value = 120ull;
+
+	ASSERT_NO_THROW(result = utils::safe_convert<TypeParam>(value));
+	EXPECT_EQ(static_cast<TypeParam>(value), result);
+}
+
+TYPED_TEST(Converters, canDoCorrectConvertionOfSigned)
+{
+	TypeParam result = 0;
+	int64_t value = 120ull;
+
+	ASSERT_NO_THROW(result = utils::safe_convert<TypeParam>(value));
+	EXPECT_EQ(static_cast<TypeParam>(value), result);
+}
+
+TYPED_TEST(Converters, cannotConvertOutOfRange)
+{
+	auto isLongDouble = typeid(TypeParam) == typeid(long double);
+
+	auto upperLimit = !isLongDouble
+	                      ? std::numeric_limits<long double>::max()
+	                      : std::numeric_limits<long double>::infinity();
+	auto lowerLimit = !isLongDouble
+	                      ? std::numeric_limits<long double>::lowest()
+	                      : -std::numeric_limits<long double>::infinity();
+
+	EXPECT_THROW(
+	    utils::safe_convert<TypeParam>(upperLimit), std::overflow_error);
+	EXPECT_THROW(
+	    utils::safe_convert<TypeParam>(lowerLimit), std::overflow_error);
+}
+
+TYPED_TEST(Converters, canConvertBool)
+{
+	TypeParam result{0};
+	ASSERT_NO_THROW(result = utils::safe_convert<TypeParam>(true));
+	EXPECT_EQ(static_cast<TypeParam>(1), result);
+
+	ASSERT_NO_THROW(result = utils::safe_convert<TypeParam>(false));
+	EXPECT_EQ(static_cast<TypeParam>(0), result);
+}
+
+TYPED_TEST(Converters, cannotConvertFloatInfinity)
+{
+	EXPECT_THROW(
+	    utils::safe_convert<TypeParam>(std::numeric_limits<float>::infinity()),
```

Comment:
float inf -> double inf should work no?

Reply:
Agree. Updated `safe_convert` and added tests to cover this case.